testdata/identical_branches.go

```diff
@@ -106,11 +106,80 @@ func identicalBranches() {
 	}
 
 	// Test confidence is reset
-	if a { // json:{"MATCH": "\"if...else if\" chain with identical branches (lines [109 111])","Confidence": 1}
+	if a { // json:{"MATCH": "\"if...else if\" chain with identical branches (lines 109 and 111)","Confidence": 1}
 		foo()
 	} else if b {
 		foo()
 	} else {
 		bar()
 	}
+
+	switch a { // MATCH /"switch" with identical branches (lines 119 and 123)/
+	// expected values
+	case 1:
+		foo()
+	case 2:
+		bar()
+	case 3:
+		foo()
+	default:
+		return newError("blah")
+	}
+
+	// MATCH:131 /"switch" with identical branches (lines 133 and 137)/
+	// MATCH:131 /"switch" with identical branches (lines 135 and 139)/
+	switch a {
+	// expected values
+	case 1:
+		foo()
+	case 2:
+		bar()
+	case 3:
+		foo()
+	default:
+		bar()
+	}
+
+	switch a { // MATCH /"switch" with identical branches (lines 145 and 147)/
+	// expected values
+	case 1:
+		foo()
+	case 3:
+		foo()
+	default:
+		if true { // MATCH /"if...else if" chain with identical branches (lines 150 and 152)/
+			something()
+		} else if true {
+			something()
+		} else {
+			if true { // MATCH /both branches of the if are identical/
+				print("identical")
+			} else {
+				print("identical")
+			}
+		}
+	}
+
+	// Skip untagged switch
+	switch {
+	case a > b:
+		foo()
+	default:
+		foo()
+	}
+
+	// Do not warn on fallthrough
+	switch a {
+	case 1:
+		foo()
+		fallthrough
+	case 2:
+		fallthrough
+	case 3:
+		foo()
+	case 4:
+		fallthrough
```

Comment:
Testing fallthrought is great.

Could you add an example with a goto ?

![goto(1).png](https://github.com/user-attachments/assets/f2475684-7d7c-4594-8041-2eab1954de1f)

I dislike them, but people can use them

Reply:
Could you please give an example with `goto`?

As I understand, for this analysis, `goto` should be considered as an ordinary statement.

```go
switch a { // the rule must warn on both branches being identical
  case 0:
    goto label
  case 1:
    goto label
}
```

Reply:
You are right. For this rule everything is fine.